compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/Fir2IrClassifierStorage.kt

```diff
@@ -161,7 +162,8 @@ class Fir2IrClassifierStorage(
         parent: IrDeclarationParent? = null,
         origin: IrDeclarationOrigin = IrDeclarationOrigin.DEFINED
     ): IrClass {
-        val descriptor = WrappedClassDescriptor()
+        val descriptor = if (regularClass is FirJavaClass) WrappedJavaClassDescriptor()
```

Comment:
If this is the only place `FirJavaClass` is used and that's why `compiler:fir:java` dependency is added, you can use `regularClass.origin == FirDeclarationOrigin.Java` instead.

Reply:
Thanks for the suggestion! Files updated.